src/arch/src/x86_64/msr.rs

```diff
@@ -282,20 +282,44 @@ pub fn create_boot_msr_entries() -> Vec<kvm_msr_entry> {
     ]
 }
 
+/// Error type for [`set_msrs`].
+#[derive(Debug, thiserror::Error)]
+pub enum SetMSRsError {
```

Comment:
My current pattern at the moment is to prefix with the function name. I tend to prefer this as it best illustrates they are bound (and this error is not used elsewhere).

Reply:
Would this mean that there would potentially be a GetMSRsError eventually as well? 

The naming here of an error class (enum) associated with each single function seems like overkill?

Reply:
> Would this mean that there would potentially be a GetMSRsError eventually as well?

Most likely.

> The naming here of an error class (enum) associated with each single function seems like overkill?

This is a somewhat opinionated area. In the Rust standard library we see instances of both function specific error types and shared error types. I lean towards specific error types as this provides better documentation and traceability. There is a balance here, if functions return mostly similar errors it may make sense to use a shared error type, otherwise function specific error types may make sense. I view error types like I view documentation here.

Reply:
Worth noting here, there wouldn't be an `GetMSRsError` in Firecracker since there is no `get_msrs` in Firecracker.